mne/beamformer/tests/test_lcmv.py

```diff
@@ -874,4 +911,72 @@ def test_lcmv_maxfiltered():
+    events = mne.find_events(raw)
+    raw.pick_types(meg='mag')
+    assert len(raw.ch_names) == 102
+    epochs = mne.Epochs(raw, events, None, preload=True)
+    data_cov = mne.compute_covariance(epochs, tmin=0.04, tmax=0.15)
+    # for now, avoid whitening to make life easier
+    noise_cov = mne.make_ad_hoc_cov(epochs.info, std=dict(grad=1., mag=1.))
+    forward = mne.read_forward_solution(fname_fwd)
+    convert_forward_solution(forward, surf_ori=True, copy=False)
+    rank = None
+    filters = make_lcmv(epochs.info, forward, data_cov, reg=reg,
+                        noise_cov=noise_cov, pick_ori=pick_ori,
+                        weight_norm=weight_norm, rank=rank)
+    _, _, _, _, G, _, _, _ = _prepare_beamformer_input(
+        epochs.info, forward, None, 'vector', noise_cov=noise_cov, rank=rank,
+        pca=False, exp=None)
+    n_channels, n_sources = G.shape
+    n_sources //= 3
+    G.shape = (n_channels, n_sources, 3)
+    G = G.transpose(1, 2, 0)  # verts, orient, ch
+    _assert_weight_norm(filters, G, weight_norm)
+
+
+def _assert_weight_norm(filters, G, weight_norm='unit-noise-gain'):
+    weights, max_power_ori = filters['weights'], filters['max_power_ori']
+    if weights.ndim == 2:  # LCMV
+        weights = weights[np.newaxis]
+        if max_power_ori is not None:
+            max_power_ori = max_power_ori[np.newaxis]
+    if max_power_ori is not None:
+        max_power_ori.shape = max_power_ori.shape + (1,)
+    if 'nsource' in filters:
+        # LCMV
+        n_sources = filters['nsource']
+        n_orient = weights.shape[1] // n_sources
+    else:
+        # DICS
+        n_orient = filters['n_orient']
+        n_sources = weights.shape[1] // n_orient
+    n_channels = weights.shape[-1]
+    for wi, w in enumerate(weights):
+        w = w.reshape(n_sources, n_orient, n_channels)
+        if filters['max_power_ori'] is not None:
+            use_G = np.sum(G * max_power_ori[wi], axis=1, keepdims=True)
+        elif filters['pick_ori'] == 'normal':
+            use_G = G[:, -1:]
+        else:
+            use_G = G
+        assert w.shape == use_G.shape == (n_sources, n_orient, n_channels)
+        got = np.matmul(w, w.conj().swapaxes(-2, -1))
+        desired = np.repeat(np.eye(n_orient)[np.newaxis], w.shape[0], axis=0)
+        if n_orient == 3 and weight_norm == 'unit-noise-gain':
+            # only the diagonal is correct
+            got = got.reshape(n_sources, -1)[:, ::n_orient + 1]
+            desired = np.ones_like(got)
+        assert_allclose(got, desired, atol=1e-7, err_msg='w @ w.conj().T')
+        # Check that the result here is a constant times a diagonal matrix
+        # (for Sekihara) or a rotated version of that (for sqrtm)
+        # got = w @ use_G.swapaxes(-2, -1)
+        # s = np.linalg.svd(got, compute_uv=False)
+        # theta = np.repeat(s[..., :1], s.shape[-1], axis=-1)
+        # assert_allclose(s, theta, rtol=1e-1, err_msg='G.T @ w = θI')
```

Comment:
@larsoner  This is not right I think - it is easier than you tested for.
`w @ G.T` should not have a constant on the diagonal, it just should be a diagonal matrix. This holds for me in the test above:
`np.allclose(np.diag(np.diag(got[0])), got[0]) `

I just did not know how to do this on our 3-dimensional matrix ....

Reply:
Last I checked it, it wasn't diagonal I don't think. But I can look early next week. If you want to try something, feel free to just make it a simple loop and I can vectorize the check later